components/experimental/src/units/measureunit.rs

```diff
@@ -129,16 +129,51 @@ impl<'data> MeasureUnitParser<'data> {
     /// Process an identifier.
     pub fn try_from_identifier(
         &self,
-        identifier: &'data str,
+        identifier: &'data [u8],
     ) -> Result<MeasureUnit, ConversionError> {
-        if identifier.starts_with('-') || identifier.ends_with('-') {
+        if identifier.starts_with(b"-") || identifier.ends_with(b"-") {
             return Err(ConversionError::InvalidUnit);
         }
 
-        let (num_part, den_part) = identifier
-            .split_once("per-")
-            .map(|(num_part, den_part)| (num_part.strip_suffix('-').unwrap_or(num_part), den_part))
-            .unwrap_or((identifier, ""));
+        /// Splits a byte slice (`haystack`) by another byte slice (`needle`).
+        /// Returns a tuple containing the part before the `needle` and the part after the `needle`.
+        ///
+        /// # Notes
+        /// - If `needle` is empty, returns the whole `haystack` and an empty slice.
+        /// - If `needle` is not found, returns the whole `haystack` and an empty slice.
+        fn split_once<'a>(haystack: &'a [u8], needle: &[u8]) -> (&'a [u8], &'a [u8]) {
```

Comment:
As discussed, merge `analyze_identifier_part` into this function and then you can avoid the multiple passes.

Reply:
done